### PR TITLE
Adding release notes for 1.17.0-beta.0 

### PR DESCRIPTION
**Pull request by bridgetkromhout**
Following the method in https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/release-notes/patch-notes.md I'm adding release notes for 1.17.0-beta.0. Please let me know if anything additional is needed for this PR! Thanks.

### CODE REVIEW

**Comment by cartyc** [comment]
Thanks @bridgetkromhout !

/ok-to-test

**Comment by saschagrunert** [comment]
I address the CI package bug in https://github.com/kubernetes-sigs/release-notes/pull/121

File: src/environments/assets.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export const assets = [
+  'assets/release-notes-1.17.0-beta.0.json',
   'assets/release-notes-1.17.0-alpha.3.json',
   'assets/release-notes-1.17.0-alpha.2.json',
```
Comment (saschagrunert) [inline src/environments/assets.ts]: Do we want to remove old pre-releases in favor of new ones? (I think this has not been discussed yet)

**Comment by saschagrunert** [comment]
@bridgetkromhout may I ask you to rebase on the latest master branch? :)

File: src/environments/assets.ts (hunk 1 above)
Comment (cartyc) [reply src/environments/assets.ts]: I was thinking once the final release was out that might be a good time to do that?

**Comment by jeefy** [comment]
I do think we should pull alpha/betas once the official release is out. The question is whether we want to do that via PRs or whether we want to have the site do it automatically.

I see positives to both approaches, but lean towards PRs. Thoughts?

**Comment by saschagrunert** [comment]
> I do think we should pull alpha/betas once the official release is out. The question is whether we want to do that via PRs or whether we want to have the site do it automatically.
> 
> I see positives to both approaches, but lean towards PRs. Thoughts?

I agree, for now we should stick to manual PRs until the automation is not completely in place.

**Comment by cartyc** [comment]
I am pro-pr for now but agree that Automation should be the end goal on that.

**Comment by bridgetkromhout** [comment]
> @bridgetkromhout may I ask you to rebase on the latest master branch? :)
Hi @saschagrunert, sorry I was on a flight without wifi and didn't get this rebased, but it looks like it got merged anyhow - appreciated!

**Pull request merged**